shared/libraries/discovery_server/src/mdnsdiscovery_server.cpp

```diff
@@ -22,27 +22,70 @@
 
 BEGIN_NAMESPACE_DISCOVERY_SERVICE
 
-MdnsDiscoveredDevice::MdnsDiscoveredDevice(const std::string& serviceName, uint32_t servicePort, const std::unordered_map<std::string, std::string>& properties)
+MdnsDiscoveredDevice::MdnsDiscoveredDevice(const std::string& serviceName, 
+                                           uint32_t servicePort, 
+                                           const std::unordered_map<std::string, std::string>& properties, 
+                                           const daq::PropertyObjectPtr& deviceInfo)
     : serviceName(serviceName)
     , servicePort(servicePort)
     , properties(std::move(properties))
+    , deviceInfo(deviceInfo)
 {
     if (this->serviceName.back() != '.')
         this->serviceName += ".";
+
+    daq::ListPtr<daq::IString> editableProps = deviceInfo.getPropertyValue("changeableProperties");
+
+    for (const auto& prop : deviceInfo.getAllProperties())
+    {
+        if (prop.getValueType() == CoreType::ctString)
+        {
+            bool found = false;
+            for (const auto& editableProp : editableProps)
+            {
+                if (prop == editableProp)
+                {
+                    found = true;
+                    break;
+                }
+            }
+            
+            if (found)
+                this->properties[prop.getName()] = (std::string)prop.getValue();
+            else
+                dynamicProperties.push_back({prop.getName(), ""});
+        }
+    }
+}
+
+size_t MdnsDiscoveredDevice::size() const
+{
+    return properties.size() + dynamicProperties.size();
+}
+
+mdns_record_t MdnsDiscoveredDevice::createRecord(const std::string& name, const std::string& value) const
+{
+    mdns_record_t record;
+    record.name = {name.c_str(), name.size()},
+    record.type = MDNS_RECORDTYPE_TXT,
+    record.data.txt.key = {name.c_str(), name.size()},
+    record.data.txt.value = {value.c_str(), value.size()},
+    record.rclass = 0,
+    record.ttl = 0;
+    return record;
 }
 
 void MdnsDiscoveredDevice::populateRecords(std::vector<mdns_record_t>& records) const 
 {
     for (const auto & [key, value] : properties)
     {
-        mdns_record_t record;
-        record.name = {serviceInstance.c_str(), serviceInstance.size()},
-        record.type = MDNS_RECORDTYPE_TXT,
-        record.data.txt.key = {key.c_str(), key.size()},
-        record.data.txt.value = {value.c_str(), value.size()},
-        record.rclass = 0,
-        record.ttl = 0;
-        records.push_back(record);
+        records.push_back(createRecord(key, value));
+    }
+
+    for (auto & [key, value] : dynamicProperties)
```

Comment:
Do we need the split into `properties` and `dynamicProperties`? All properties could be in 1 map with the check done here:

```
    for (const auto & [key, value] : properties)
    {
        if (dynamicProperties.count(key))
            records.push_back(createRecord(key, deviceInfo.getPropertyValue(key));
        else
             records.push_back(createRecord(key, value));
    }
```